code/datums/components/waddling.dm

```diff
@@ -21,6 +21,8 @@
 /datum/component/waddling/proc/Waddle()
 	SIGNAL_HANDLER
 
-	animate(parent, pixel_z = 4, time = 0)
-	animate(pixel_z = 0, transform = turn(matrix(), pick(-12, 0, 12)), time=2)
-	animate(pixel_z = 0, transform = matrix(), time = 0)
+	var/rot_degrees = pick(-12, 0, 12)
+	var/mob/living/L = parent
```

Comment:
This component isn't only made for living

Reply:
I’m not sure why I put that typecast there, I’ll get rid of it when I’m back at my computer.

Reply:
I changed the typecast to just atom. Anything less broad an transform won't be a member of the class anymore.

Reply:
This cast isn't needed

Reply:
parent is a datum, which doesn't have the transform variable, it needs to be typecasted to an atom to index transform for the rotations.

Reply:
Then cast it to movable

Reply:
It's a movable now.